src/EPR.Calculator.Frontend/Controllers/LocalAuthorityUploadFileErrorController.cs

```diff
@@ -72,7 +72,7 @@ public async Task<IActionResult> Upload(IFormFile fileUpload)
             var localAuthorityDisposalCostsValues = await CsvFileHelper.PrepareLapcapDataForUpload(fileUpload);
 
             this.ViewData["localAuthorityDisposalCosts"] = localAuthorityDisposalCostsValues.ToArray();
-            this.TempData["LapcapFileName"] = fileUpload.FileName;
+            this.HttpContext.Session.SetString("LapcapFileName", fileUpload.FileName);
```

Comment:
Please create constants for session variables
Also, you need to clean up all the values in try/catch block. Thanks